EIPS/eip-3416.md

```diff
@@ -0,0 +1,114 @@
+---
+eip: 3416
+title: Median gas premium and miner fees
+author: HexZorro (@hexzorro), strongly based on EIP-1559 circa Aug-2020.
+discussions-to: https://ethereum-magicians.org/t/eip-3416-median-gas-premium/5755
+status: Draft
+type: Standards Track
+category: Core
+created: 2021-03-18
+---
+
+## Simple Summary
+
+A transaction pricing mechanism that includes a fixed-per-block network fee that is *not*-burned plus a *median* premium fee per transaction. Also, dynamically expands/contracts block sizes to deal with transient congestion. We target the following goals:
+
+* Gas prices spikes are mathematically smoothed out.
+* Maintain gas price preference, i.e. transaction senders willing to pay extra in fees will be rewarded with early preferential inclusion in the blocks, because the miners want to maximize their profits and include transactions with higher fee caps first to maximize the median.
+* Final gas price paid by the sender is, most of the time, smaller than the maximum gas price specified by sender. 
+* Gas pricing is more robust to sender manipulation or miner manipulation.
+* All fees go to miners.
+
+## Abstract
+
+There is a base fee per gas in protocol, which can move up or down by a maximum of 1/8 in each block. The base fee per gas is adjusted by the protocol to target an average gas usage per block instead of an absolute gas usage per block.  The base fee is increased when blocks are over the gas limit target and decreases when blocks are under the gas limit target. The base fee per gas is *not burned*. Transaction senders specify their fees by providing *only one value*:
```

Comment:
```suggestion
There is a base fee per gas in protocol, which can move up or down by a maximum of 1/8 in each block. The base fee per gas is adjusted by the protocol to target an average gas usage per block instead of an absolute gas usage per block.  The base fee is increased when blocks are over the gas limit target and decreases when blocks are under the gas limit target. Transaction senders specify their fees by providing *only one value*:
```
Generally speaking, EIPs should focus on what they do rather than what they don't do since the former set is small and the latter set is infinite.